sys/net/application_layer/dhcpv6/client.c

```diff
@@ -75,9 +87,11 @@ static uint8_t send_buf[DHCPV6_CLIENT_SEND_BUFLEN];
 static uint8_t recv_buf[DHCPV6_CLIENT_BUFLEN];
 static uint8_t best_adv[DHCPV6_CLIENT_BUFLEN];
 static uint8_t duid[DHCPV6_CLIENT_DUID_LEN];
+static addr_lease_t addr_leases[CONFIG_DHCPV6_CLIENT_ADDR_LEASE_MAX];
 static pfx_lease_t pfx_leases[CONFIG_DHCPV6_CLIENT_PFX_LEASE_MAX];
 static server_t server;
-static event_timeout_t solicit_renew_timeout, rebind_timeout;
+static event_timeout_t solicit_renew_timeout, rebind_timeout,
+                       deprecate_timeout, remove_timeout;
```

Comment:
I might be mistaken, but since deprecation timeout always happens before removal timeout, you might be able to safe some RAM here, by only using one timeout event for both event types.

Reply:
I tried to refactor this in 2ec46684d285a6be3e78956abd46193cc3ea51d6. I hope this works :)